core/server/src/main/java/alluxio/master/FaultTolerantAlluxioMaster.java

```diff
@@ -98,11 +97,13 @@ public void start() throws Exception {
 
           // When transitioning from master to standby, recreate the masters with a read-only
```

Comment:
I agree, it would be better to re-create so that we don't need to work about leftover state. I've updated the PR to re-create masters. This required a nontrivial amount of refactoring, so @jsimsa you should take another look